gRPC/rust-sdks/cosmwasm_client/src/cosmwasm_connector.rs

```diff
@@ -0,0 +1,105 @@
+use crate::pull_service::PullResponseCosmWasm;
+use cosmrs::cosmwasm::MsgExecuteContract;
+use cosmrs::rpc::Client;
+use cosmrs::proto::cosmos::auth::v1beta1::query_client::QueryClient;
+use cosmrs::proto::cosmos::auth::v1beta1::{BaseAccount, QueryAccountRequest};
+use cosmrs::proto::traits::Message;
+use cosmrs::{
+    crypto::secp256k1,
+    rpc,
+    tx::{self, AccountNumber, Fee, Msg, SignDoc, SignerInfo},
+    AccountId, Coin,
+};
+use serde_json::json;
+use std::str::FromStr;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum CosmWasmConnectorError {
+    #[error("invalid grpc response")]
+    InvalidGRPCResponse,
+}
+
+pub async fn invoke_cosmwasm_chain(cosmwasm: PullResponseCosmWasm) {
+    const CHAIN_ID: &str = "osmo-test-5";
+    const ACCOUNT_PREFIX: &str = "osmo";
+    const DENOM: &str = "uosmo";
+    const COIN_AMOUNT: u128 = 10000;
+    const GAS_LIMIT: u64 = 900000;
+
+    let rpc_url = "<RPC URL>"; // Rpc url for desired chain "https://rpc.testnet.osmosis.zone:443"
+    let secret_key = "<PRIVATE KEY>"; // Your Private Key
+    let contract_address = "<CONTRACT ADDRESS>"; // Address of your smart contract
+
+    let sender_private_key =
+        secp256k1::SigningKey::from_slice(hex::decode(secret_key).unwrap().as_slice()).unwrap();
+
+    let sender_public_key = sender_private_key.public_key();
+    let sender_account_id = sender_public_key.account_id(ACCOUNT_PREFIX).unwrap();
+    println!("osmosis account: {:?}", sender_account_id);
+
+    let (account_number,sequence_number) = collect_acc_sequence_number(rpc_url.to_string(),sender_account_id.clone());
```

Comment:
`collect_acc_sequence_number` is an **async** function and returns a Result, so you have to handle it accordingly